opm/simulators/wells/RateConverter.hpp

```diff
@@ -855,6 +855,191 @@ namespace Opm {
             Details::RegionAttributes<RegionId, Attributes> attr_;
 
         };
+
+        /**
+         * Computes hydrocarbon weighed average pressures over regions
+         *
+         * \tparam FluidSystem Fluid system class. Expected to be a BlackOilFluidSystem
+         *
+         * \tparam Region Type of a forward region mapping.  Expected
+         * to provide indexed access through \code operator[]()
+         * \endcode as well as inner types \c value_type, \c
+         * size_type, and \c const_iterator.  Typically \code
+         * std::vector<int> \endcode.
+         */
+        template <class FluidSystem, class Region>
+        class AverageRegionalPressure {
+        public:
```

Comment:
The file name (and type) RateConverter is now misleading since it used to do more stuff. I suggest renaming to AverageRegionalCalculator. Any objections?

Reply:
> I suggest renaming to AverageRegionalCalculator. Any objections?

I object.  That's not a good name as the primary purpose is **still** to convert flow rates from surface conditions to reservoir conditions (and back).

Reply:
The AverageRegionalCalculator is a new class,  the old SurfaceToReservoirVoidage is still there to do the conversion of the flow rates. What I want is to rename the file.

Reply:
I think we should split these classes into separate files. It is good to have one important class with unique function per file. Maybe also put common facilities in its own, separate file.

Reply:
I would still argue for keeping them in the same file and instead rename the file. The implementation are almost the same. But since both @bska and @atgeirr thinks differently I will yield and make a new file called RegionalAverageCalculator.hpp. Can I put the facilities there? If not do you have a suggestion for a name for the file with the facilities? FacilitiesForRegionalAverageCalculators?